### ISSUE TITLE
RealmMigrationNeededException only on Android 4.2.2 and Samsung Galaxy S3 Mini

### ISSUE THREAD

**Issue by deva666** [realm/realm-java#4306]
Our crash logs just start showing up this crash.
It only affects Android version 4.2.2 and Samsung GT-I8200N phone.
Realm version 2.3.1
```
java.lang.RuntimeException: Unable to start activity ComponentInfo{MainActivity}: io.realm.exceptions.RealmMigrationNeededException: The 'RealmPumpBasalRateConfig' class is missing from the schema for this Realm.
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2262)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2316)
	at android.app.ActivityThread.access$700(ActivityThread.java:158)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1296)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:176)
	at android.app.ActivityThread.main(ActivityThread.java:5365)
	at java.lang.reflect.Method.invokeNative(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:511)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1102)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:869)
	at dalvik.system.NativeStart.main(Native Method)
Caused by: io.realm.exceptions.RealmMigrationNeededException: The 'RealmPumpBasalRateConfig' class is missing from the schema for this Realm.
	at io.realm.RealmPumpBasalRateConfigRealmProxy.validateTable(RealmPumpBasalRateConfigRealmProxy.java:495)
	at io.realm.DefaultRealmModuleMediator.validateTable(DefaultRealmModuleMediator.java:66)
	at io.realm.Realm.initializeRealm(Realm.java:348)
	at io.realm.Realm.createAndValidate(Realm.java:313)
	at io.realm.Realm.createInstance(Realm.java:278)
	at io.realm.RealmCache.createRealmOrGetFromCache(RealmCache.java:143)
	at io.realm.Realm.getDefaultInstance(Realm.java:209)
```

**Comment by kneth** [comment]
How easy is it to reproduce the issue? I don't have GT-I8200N at the office but I have GT-I8190 and SM-T111, so if you send an APK to help@realm.io I be happy to try on those devices. If you can provide source code, it would be it easier for us to figure out what the cause is.

**Comment by deva666** [comment]
Reproducing it is easy. Just start the app, create a new user and then the crash happens. 
We can not give our source code, I can only give code snippets (how we create Realm configurations, for example).
The app can be downloaded from Playstore https://play.google.com/store/apps/details?id=com.mysugr.android.companion
I can also create a debug build and give you debuggable APK if it helps.
Thanks.

**Comment by kneth** [comment]
Yes, please send us a debug version so we can try it on our test devices.

**Comment by deva666** [comment]
OK, debug APK sent to help@realm.io

**Comment by kneth** [comment]
@deva666 I have tried the app on Samsung SM-T111 (Android 4.2.2) and Samsung I-8190N (Android 4.1.2) and I experienced no crashes.

**Comment by kneth** [comment]
I have not been able to find an I-8200N. @deva666 Can you reproduce it on other devices?

**Comment by deva666** [comment]
I see in the crash logs that these devices are failing (all only on Android 4.2.2)
SAMSUNG SM-T111
SAMSUNG SM-T111M
SAMSUNG GT-I8200N
SAMSUNG SM-T110

**Comment by Zhuinden** [comment]
Amazing. This ties into https://github.com/realm/realm-java/issues/3651 and will most likely be fixed by https://github.com/realm/realm-java/pull/4402 (native code is bugged on samsung devices as per https://code.google.com/p/android/issues/detail?id=81692)

**Comment by deva666** [comment]
Yes, I also see one crash on Wiko WAX (Android 4.3) as also mentioned in the [#3651](https://github.com/realm/realm-java/issues/3651)

**Comment by kneth** [comment]
It is unlikely that the vendors will ever patch these old versions on Android. We hope that we are able to provide a workaround in #4402 (other libraries can do the same).

**Comment by kneth** [comment]
@deva666 Until #4402 you can find a build in #3651. We will be happy if you could give it a try as we only have a limited number (one!) device which can reproduce it.

**Comment by bmeike** [comment]
@deva666 
We've merged #4402 which we believe will fix this problem.  I am closing this ticket.  By all means, please re-open it if the problem persists.
